rts/idris_utf8.c

```diff
@@ -78,6 +78,36 @@ unsigned idris_utf8_index(char* s, int idx) {
    return top;
 }
 
+char* idris_utf8_advance(char* str, int i) {
```

Comment:
@melted I don't think that will quite work - it cuts off the end of the character because i hits 0 at the beginning of the multibyte sequence. But I'll do something similar.